lib/streamlit/elements/markdown.py

```diff
@@ -271,6 +277,96 @@ def divider(self) -> DeltaGenerator:
         divider_proto.element_type = MarkdownProto.Type.DIVIDER
         return self.dg._enqueue("markdown", divider_proto)
 
+    @gather_metrics("badge")
+    def badge(
+        self,
+        label: str,
+        *,  # keyword-only arguments:
+        icon: str | None = None,
+        color: Literal[
+            "blue",
+            "green",
+            "orange",
+            "red",
+            "violet",
+            "gray",
+            "grey",
+            "rainbow",
+            "primary",
+        ] = "blue",
+    ) -> DeltaGenerator:
+        """Display a colored badge with an icon and label.
+
+        You can also insert badges directly in Markdown, e.g. via
+        `st.markdown(":blue-badge[Home]")`. This works in all places where Streamlit
+        supports Markdown, e.g. widget labels or `st.table` cells. See `st.markdown`
+        for more information.
+
+        Parameters
+        ----------
+        label : str
+            The label to display in the badge. The label can optionally contain
+            GitHub-flavored Markdown of the following types: Bold, Italics,
```

Comment:
@sfc-gh-dmatthews FYI there's not really any good explanation online about which Markdown types are supported inside the `remark-directive` plugin, which we use here. So I just tried out a bunch of types and added the ones that worked here. Also note that unlike other Markdown features, this doesn't unwrap unsupported types.